src/modules/services_manager_service.rb

```diff
@@ -232,67 +222,63 @@ def errors
 
     # Activates the service in cache
     #
-    # @param [String] service name
+    # @param name [String] service name
     # @return [Boolean]
-    def switch(service)
-      active(service) ? deactivate(service) : activate(service)
+    def switch(name)
+      active(name) ? deactivate(name) : activate(name)
     end
 
     # Resets service changes
     #
-    # @param [String] service name
+    # @param name [String] service name
     # @return [Boolean]
-    def reset_service(service)
-      exists?(service) do
-        find(service).reset
+    def reset_service(name)
+      exists?(name) do |service|
+        service.reset
         true
       end
     end
 
     # Sets start_mode for a service (in memory only, use save())
     #
-    # @param service [String] service name
+    # @param name [String] service name
     # @param mode    [Symbol] Start mode
     # @see Yast::SystemdServiceClass::Service#start_modes
-    def set_start_mode(service, mode)
-      exists?(service) do
-        find(service).start_mode = mode
-      end
+    def set_start_mode(name, mode)
+      exists?(name) { |s| s.start_mode = mode }
     end
 
     # Returns the start_mode for the given service (from memory)
     #
+    # @param name [String] service name
     # @return [Symbol] Start mode
-    def start_mode(service)
-      exists?(service) do
-        find(service).start_mode
-      end
+    def start_mode(name)
+      exists?(name) { |s| s.start_mode }
     end
 
     # Returns the list of supported start modes for the given service
     #
+    # @param name [String] service name
     # @return [Array<Symbol>] Supported start modes
```

Comment:
this a bit scary me. exists can return false so not documented. In general I do not like much idea of that exists in each method which result that any method can return false. It will be much easier to catch exception than to check every response that it has not been false.

Reply:
Yes, I do not like that `exists?` *thingie* to be honest (it is basically old code). We can plan to change it in the future, but out of the scope of this PR.